presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -323,36 +327,53 @@ public synchronized HivePageSinkMetadata generatePageSinkMetadata(MetastoreConte
 
     public synchronized void createDatabase(MetastoreContext metastoreContext, Database database)
     {
-        setExclusive((delegate, hdfsEnvironment) -> delegate.createDatabase(metastoreContext, database));
+        setExclusive((delegate, hdfsEnvironment) -> {
+            delegate.createDatabase(metastoreContext, database);
+            return INSTANCE;
```

Comment:
`return EMPTY_HIVE_COMMIT_HANDLE`. Same for all other places in the class. This is why we don't recommend have INSTANCE in SPI; rather, it should be declared inside each connector.